src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -413,6 +414,42 @@ def connection_is_activated(self, connection_id):
 
         return connection_is_activated
 
+    def get_default_gateway_state(self):
+        """Gets information about the current default gateway route. Returns a dictionary
+        including the gateway address, the interface name associated with the address, and
+        the NetworkManager connection associated with that device.
+        """
+
+        # TODO: broadcast signal if:
+        #   the default gateway IP changes
+        #   the interface associated with the default gateway changes
+        #   the connection associated with the interface of the default gateway changes
+
+        # to get an interface name from the default gatway:
+        # with pyroute2.IPRoute() as ipr:
+        #   ipr.get_links()[ipr.get_default_routes()[0]['attrs'][3][1] - 1]['attrs'][0][1]
+
+        gateway_state = None
+
+        with pyroute2.IPRoute() as ipr:
+            default_routes = ipr.get_default_routes()
+            if default_routes:
+                gateway_state = {'address': None, 'interface': None, 'connection_id': None}
+                default_route = default_routes[0]
+                gateway_state['address'] = default_route['attrs'][2][1]
+                # OIF is the output interface associated with the route, conveniently
+                # represented in an index value that's off by one.
+                default_route_oif = default_route['attrs'][3][1]
+                oif_device_attrs = ipr.get_links()[default_route_oif - 1]['attrs']
+                gateway_state['interface'] = oif_device_attrs[0][1]
+
```

Comment:
Everything before here really doesn't have anything to do with NetworkManager. It should be located in netcheck.py or another file.

Reply:
I put it here because it does things that we decided were the domain of NetworkManagerHelper. Namely, it iterates over devices and figures out the relationship between devices, connections, and interface names.

Reply:
We discussed this in person. You agreed to move everything above line 445 into (probably) netcheck.py. Everything after line 445 will be put into a method that returns the connection_id based on the interface name.